File: src/main/java/no/nav/syfo/utils/OIDCUtil.java
package no.nav.syfo.utils;

import no.nav.syfo.services.ws.OnBehalfOfOutInterceptor;
import no.nav.security.oidc.OIDCConstants;
import no.nav.security.oidc.context.OIDCRequestContextHolder;
import no.nav.security.oidc.context.OIDCValidationContext;
import org.apache.cxf.endpoint.Client;

public class OIDCUtil {

    public static String tokenFraOIDC(OIDCRequestContextHolder contextHolder, String issuer) {
        OIDCValidationContext context = (OIDCValidationContext) contextHolder
                .getRequestAttribute(OIDCConstants.OIDC_VALIDATION_CONTEXT);
        return context.getToken(issuer).getIdToken();
    }

    public static void leggTilOnBehalfOfOutInterceptorForOIDC(Client client, String OIDCToken) {
        client.getRequestContext().put(OnBehalfOfOutInterceptor.REQUEST_CONTEXT_ONBEHALFOF_TOKEN_TYPE, OnBehalfOfOutInterceptor.TokenType.OIDC);
        client.getRequestContext().put(OnBehalfOfOutInterceptor.REQUEST_CONTEXT_ONBEHALFOF_TOKEN, OIDCToken);
    }
}

File: src/main/kotlin/no/nav/syfo/oidc/OIDCUtil.kt
package no.nav.syfo.oidc

import no.nav.security.oidc.OIDCConstants
import no.nav.security.oidc.context.OIDCRequestContextHolder
import no.nav.security.oidc.context.OIDCValidationContext
import no.nav.syfo.services.ws.OnBehalfOfOutInterceptor
import org.apache.cxf.endpoint.Client

object OIDCUtil {
    @JvmStatic
    fun tokenFraOIDC(contextHolder: OIDCRequestContextHolder, issuer: String?): String {
        val context = contextHolder
            .getRequestAttribute(OIDCConstants.OIDC_VALIDATION_CONTEXT) as OIDCValidationContext
        return context.getToken(issuer).idToken
    }

    @JvmStatic
    fun leggTilOnBehalfOfOutInterceptorForOIDC(client: Client, OIDCToken: String?) {
        client.requestContext[OnBehalfOfOutInterceptor.REQUEST_CONTEXT_ONBEHALFOF_TOKEN_TYPE] = OnBehalfOfOutInterceptor.TokenType.OIDC
        client.requestContext[OnBehalfOfOutInterceptor.REQUEST_CONTEXT_ONBEHALFOF_TOKEN] = OIDCToken
    }
}
